src/subscribe.ts

```diff
@@ -1,19 +1,23 @@
-import { Sink } from './types';
+import { Consumer, Talkback } from './types';
 
-export type Observer<T> = {
-  next: (data: T) => void;
-  error?: (err: any) => void;
-  complete?: () => void;
-};
-
-export function subscribe<T>(o: Observer<T>): Sink<T> {
+export function subscribe<T>(
```

Comment:
I would still prefer the old interface here. Especially one that has seperate error and complete functions

Reply:
It also means that this PR has to touch a lot of code that is not really needed

Reply:
We need to decide one way of the other way. We still have time for breaking changes since we barely started, so it's not a problem that we are touching a lot of code (we are also touching a lot of types! see e.g. `Producer`). So let's put the "touch a lot of code" concern aside.

I know you prefer the observer object for clarity, but ergonomics or clarity is not a goal for this library. Performance and lightweightness is. Plus, I also realized some problems with the separate error and complete functions: they don't accurately describe how callbags work.
Callbags *terminate* with `2` always, and termination can either be implicitly successful or explicitly failing. A `subscribe` means you're interested in two cases: `1` or `2`.

Note that nothing stops you from making a separate function that has the observer object, and maybe it could even be in this library? But by default we should pick the option that has the least overhead possible. At least in this library.

Reply:
I guess I will use subscribe not often enough to warrant a big discussion.